runtime/major_gc.c

```diff
@@ -1322,28 +1322,42 @@ static intnat ephe_mark (intnat budget, uintnat for_cycle,
     }
     budget -= Whsize_wosize(i);
 
-    if (force_alive || alive_data) {
-      if (data != caml_ephe_none && Is_block(data)) {
-        caml_darken (domain_state, data, 0);
-      }
-      Ephe_link(v) = domain_state->ephe_info->live;
-      domain_state->ephe_info->live = v;
-      *prev_linkp = todo;
+    bool keep;
```

Comment:
I guess I'd initialize this to `true` here, with the comment from line 1339, then lose the `else` block at 1338-1341. Also maybe call it `leave`?

Reply:
Since C compiler warnings these days will reliably complain about use of uninitialised variables, I prefer to declare this kind of thing uninitialised so that the compiler verifies I picked a value for it along every reachable path.